tests/testthat/test-linter_tags.R

```diff
@@ -79,11 +79,13 @@ test_that("warnings occur only for deprecated linters", {
 
 test_that("available_linters matches the set of linters available from lintr", {
   lintr_db <- available_linters(exclude_tags = NULL)
-  linters_in_namespace <- setdiff(ls(asNamespace("lintr"), pattern = "_linter$"), "is_linter")
+  defunct_linters <- subset(read.csv(system.file("lintr", "linters.csv", package = "lintr")), tags == "defunct")$linter
```

Comment:
`lintr_db$linter[vapply(lintr_db$tags, `%in%`, logical(1L), y = "defunct")]`
avoids re-reading the CSV.